app/src/main/kotlin/com/saltedge/authenticator/features/security/LockableActivity.kt

```diff
@@ -69,12 +69,13 @@ abstract class LockableActivity :
         override fun showLockWarning() {
             val snackbar = this@LockableActivity.buildWarning(
                 getString(R.string.warning_application_was_locked),
-                snackBarDuration = Snackbar.LENGTH_LONG
+                snackBarDuration = 5000,
+                actionResId = R.string.actions_cancel
             )
             snackbar?.addCallback(object : Snackbar.Callback() {
                 override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
                     super.onDismissed(transientBottomBar, event)
-                    presenter.onSnackbarDismissed()
+                    if (event == DISMISS_EVENT_TIMEOUT) presenter.onSnackbarDismissed()
```

Comment:
only timeout should be initiator of screen-lock

Reply:
I think this is the best option other possible, largely due to its simplicity.
If you have other more interesting ideas, welcome.